dashboard/app/models/unit_group.rb

```diff
@@ -195,7 +195,8 @@ def update_scripts(new_units, alternate_units = nil)
     units_to_remove -= alternate_units.map {|hash| Script.find_by_name!(hash['alternate_script'])}
 
     if units_to_remove.any?(&:prevent_course_version_change?)
-      raise 'Cannot remove units that have resources or vocabulary'
+      unit_names = units_to_remove.select(&:prevent_course_version_change?).map(&:name)
```

Comment:
Could this be moved out of the if block? So instead do something like:
```
unit_names = units_to_remove.select(&:prevent_course_version_change?).map(&:name)
raise "Cannot remove units that have resources or vocabulary: #{unit_names}" if unit_names.length > 0
```
(`unit_names` might need to be renamed in this case)